doc/website/release-notes/iceoryx-unreleased.md

```diff
@@ -117,6 +117,7 @@
 - Remove `algorithm::uniqueMergeSortedContainers` from `algorithm.hpp`
 - Move `std::string` conversion function to `iceoryx_dust` [\#1612](https://github.com/eclipse-iceoryx/iceoryx/issues/1612)
 - The posix call `unlink` is directly used in `UnixDomainSocket` [\#1622](https://github.com/eclipse-iceoryx/iceoryx/issues/1622)
+- Make iox::byte_t a distinct type [\#1900](https://github.com/eclipse-iceoryx/iceoryx/issues/1900)
```

Comment:
@zmostafa I updated #1900 and added three tasks to the issue. Two of them will be accomplished by this PR and the last one has to wait until we can switch to C++17. Since the issue won't be closed, please remove this line.